R/utils-shared.R

```diff
@@ -448,16 +448,28 @@ getWaterDataUrl <- function(data) {
 #' @param data A list of gsplot objects to display on the plot.
 #' @return gsplot object with approval bar rectangle styles applied.
 ApplyApprovalBarStyles <- function(object, data) {
+  ylim <- ylim(object)$side.2
+  ylog <- object$global$par$ylog
+
+  if (ylim[1] == ylim[2]) {
+    # Cope with the rare case of the time series plot being a horizontal line,
+    # in which case we have to preemptively compensate for some y-axis interval
+    # defaulting code inside R graphics. The 40% factor here comes from the R
+    # source code, last seen at 
+    # http://docs.rexamine.com/R-devel/Rgraphics_8h.html#a5233f80c52d4fd86d030297ffda1445e
+    if (!isEmptyOrBlank(ylog) && ylog) {
+      ylim <- c(10^(0.6 * log10(ylim[1])), 10^(1.4 * log10(ylim[2])))
+    }
+    else {
+      ylim <- c(0.6 * ylim[1], 1.4 * ylim[2])
+    }
+  }
```

Comment:
It seems like we should have a calculation of y top and y bottom with buffer & reverse be a standalone function since it appears in `RescaleYTop`,  `ApprovalBarY`, and now here. I'm going to add that to tech debt.